File: app/src/main/java/com/fastaccess/provider/timeline/handler/SubScriptHandler.java
package com.fastaccess.provider.timeline.handler;

import android.text.SpannableStringBuilder;
import android.text.style.RelativeSizeSpan;
import android.text.style.SubscriptSpan;

import net.nightwhistler.htmlspanner.TagNodeHandler;

import org.htmlcleaner.TagNode;

public class SubScriptHandler extends TagNodeHandler {

    @Override public void handleTagNode(TagNode node, SpannableStringBuilder builder, int start, int end) {
        builder.setSpan(new SubscriptSpan(), start, end, 33);
        builder.setSpan(new RelativeSizeSpan(0.8f), start, end, 33);
    }
}

File: markdown/src/main/java/com/fastaccess/markdown/spans/SubScriptHandler.kt
package com.fastaccess.markdown.spans

import android.text.SpannableStringBuilder
import android.text.style.RelativeSizeSpan
import android.text.style.SubscriptSpan
import net.nightwhistler.htmlspanner.SpanStack
import net.nightwhistler.htmlspanner.TagNodeHandler
import org.htmlcleaner.TagNode

class SubScriptHandler : TagNodeHandler() {
    override fun handleTagNode(node: TagNode?, builder: SpannableStringBuilder?, start: Int, end: Int, spanStack: SpanStack) {
        builder?.let {
            builder.setSpan(SubscriptSpan(), start, end, 33)
            builder.setSpan(RelativeSizeSpan(0.8f), start, end, 33)
        }
    }
}